src/core/hooks/index.ts

```diff
@@ -1,3 +1,11 @@
+import type { AppDispatch, RootState } from '../store';
+import { TypedUseSelectorHook, useDispatch, useSelector } from 'react-redux';
+
 export { default as useApiClient } from './useApiClient';
 export { default as useEnv } from './useEnv';
 export { default as useNumericRouteParams } from './useNumericRouteParams';
+
+// Use throughout instead of plain `useDispatch` and `useSelector`
+type DispatchFunc = () => AppDispatch;
+export const useAppDispatch: DispatchFunc = useDispatch;
```

Comment:
I'm curious about this, I guess we are doing our "custom" dispatch function because we have to type it?

Reply:
yes! the og redux dispatch function is not typed to our data, and it also has no idea if we use middleware, so even if we are not dispatching middleware things now, it felt like it could not hurt to get more typing into it and prepare for a potential future. This is the text i based this on: https://redux.js.org/usage/usage-with-typescript#define-typed-hooks